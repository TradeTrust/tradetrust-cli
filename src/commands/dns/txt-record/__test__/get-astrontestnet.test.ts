import { handler } from "../get";

describe("get", () => {
  afterEach(() => {
    jest.clearAllMocks();
  });
  it("should return dns-txt", async () => {
    const consoleSpy = jest.spyOn(console, "table");
    await handler({ location: "dev-astronlayer2.bitfactory.cn" });

    const mockCall1 = consoleSpy.mock.calls[0][0];
    mockCall1.sort((a: any, b: any) => {
      if (a.netId < b.netId) return -1;
      if (a.netId > b.netId) return 1;
      if (a.addr < b.addr) return -1;
      if (a.addr > b.addr) return 1;
      return 0;
    });
    // dns-txt
    expect(mockCall1).toMatchSnapshot();
  });
});
