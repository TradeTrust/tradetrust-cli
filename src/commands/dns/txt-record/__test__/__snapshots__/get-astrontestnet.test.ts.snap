// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`get should return dns-txt 1`] = `
[
  {
    "addr": "0xb1Bf514b3893357813F366282E887eE221D5C2dA",
    "dnssec": false,
    "net": "ethereum",
    "netId": "21002",
    "type": "openatts",
  },
  {
    "addr": "0xdAEe89A37fEEBCBFAc94aBA63Fb163808dAc38Fb",
    "dnssec": false,
    "net": "ethereum",
    "netId": "21002",
    "type": "openatts",
  },
]
`;
